File: voting-backend/src/main/java/ch/rafaelurben/tamtour/voting/dto/admin/VotingSetDto.java
package ch.rafaelurben.tamtour.voting.dto.admin;

import ch.rafaelurben.tamtour.voting.dto.VotingPositionMapDto;
import ch.rafaelurben.tamtour.voting.dto.VotingUserDto;
import java.time.OffsetDateTime;

public record VotingSetDto(
    Long id,
    VotingUserDto votingUser,
    OffsetDateTime createdAt,
    OffsetDateTime modifiedAt,
    OffsetDateTime submittedAt,
    boolean submitted,
    boolean disqualified,
    VotingPositionMapDto positionMap) {}

File: voting-frontend/src/app/dto/admin/voting-set.dto.ts
import { VotingUserDto } from '../voting-user.dto';
import { VotingPositionMapDto } from '../voting-position-map.dto';

export interface VotingSetDto {
  id: number;
  votingUser: VotingUserDto;
  createdAt: string; // ISO datetime string including timezone
  modifiedAt: string; // ISO datetime string including timezone
  submittedAt: string | null; // ISO datetime string including timezone
  submitted: boolean;
  disqualified: boolean;
  positionMap: VotingPositionMapDto;
}
